flexget/plugins/output/slack.py

```diff
@@ -0,0 +1,120 @@
+    schema = {
+        'type': 'object',
+        'properties': {
+            'webhook-url': {'type': 'string'},
+            'text': {'type': 'string', 'default': default_body},
+            'channel': {'type': 'string'},
+            'username': {'type': 'string'},
+            'icon-emoji': {'type': 'string'},
+        },
+        'required': ['webhook-url'],
+        'additionalProperties': False
+    }
+
+    @plugin.priority(0)
+    def on_task_output(self, task, config):
+
+        for entry in task.accepted:
+            webhook_url = config.get('webhook-url')
+            text = config.get('text')
+            channel = config.get('channel')
+            username = config.get('username')
+            icon_emoji = config.get('icon-emoji')
+
+            if icon_emoji:
+                icon_emoji = ":%s:" % icon_emoji
+
+            try:
+                text = entry.render(text)
+            except RenderError as e:
+                log.warning('Problem rendering `text`: %s' % e)
+                text = 'Download started'
+
+            self.send_post(task, webhook_url, text, channel, username,
+                           icon_emoji)
+
+    def send_post(self, task, webhook_url, text, channel, username,
+                  icon_emoji):
+
+        data = {'text': text}
+        if channel:
+            data['channel'] = channel
+        if username:
+            data['username'] = username
+        if icon_emoji:
+            data['icon_emoji'] = icon_emoji
+
+        if task.options.test:
+            log.info('Test mode. Slack notification would be:')
+            log.info('    Webhook URL: {}'.format(webhook_url))
+            log.info('    Text: {}'.format(text))
+            if channel:
+                log.info('    Channel: {}'.format(channel))
+            if username:
+                log.info('    Username: {}'.format(username))
+            if icon_emoji:
+                log.info('    Icon Emoji: :{}:'.format(icon_emoji))
+            log.info('    Raw POST Data: {}'.format(json.dumps(data)))
+            # Early return (test mode)
+            return
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'User-Agent': 'Flexget'
+        }
+        try:
+            response = task.requests.post(webhook_url, headers=headers,
+                                          data=json.dumps(data),
+                                          raise_status=False)
+        except Exception as e:
+            log.error('Unable to connect to Slack API: {}'.format(e))
+            return
+
+        response_code = response.status_code
+
+        if response_code == 200:
+            log.debug('Slack notification sent')
+        elif response_code == 500:
+            log.warning('Slack notification failed: server problem')
+        elif response_code >= 400:
+            log.error('Slack API error {}: {}'.format(response_code, response.content))
```

Comment:
Please use index numbers inside string format params. This is not compatible with python 2.6. Not a huge deal but easily solvable